### DISCUSSION TITLE
Is the locator thread safe？

### DISCUSSION THREAD

**Discussion by martinrhan** [reactiveui/splat#984] [Q&A] [answered]
I would like to register service to Locator.CurrentMutable on difference thread concurrently, does it has internal thread safe mechanism, or I need to use a lock?

**Comment by ChrisPulman** [comment, marked as answer]
Hi @martinrhan I would recommend using a lock to ensure thread safety.

**Reply by ChrisPulman** [reply]
FYI: The internal mechanism is using a standard dictionary rather than a concurrent dictionary to ensure platform compatibility.

**Reply by anaisbetts** [reply]
This Sounds Bad, but usually all of the cross-thread accesses of the Locator are read-only - i.e. you initialize your Locator on startup, then all of the Gets are Fine because they're only reading the dictionary. I would honestly recommend OP just initialize the locator on a single thread.
